_posts/2018-02-28-Update-to-Java11.md

```diff
@@ -0,0 +1,124 @@
+
+It will not work for method/constructor parameters or return types, member variables/fields, or catch formals. Only for logic within methods, including loop indexes.
+
+Of course, we still have all Java 9 goodies such as new Stream functions:
+
+```
+jshell> IntStream.range(0,10).takeWhile(x -> x < 5).forEach(System.out::println)
+0
+1
+2
+3
+4
+```
+
+If you update to any of these Java versions, 9, 10, 11, you get access to JShell and goodies like that. JShell that was introduced in Java 9, starts up now significantly faster, making it much better REPL experience for those who value it.
+
+There is some Docker love coming up in Java 10 - improved container detection and resource configuration usage. So Java becomes more aware of its environment, especially running in containers.
+
+As always, better performance, improved garbage collection, experimental Jit compiler, open sourced root certificates, yada yada yada. Expect the major releases to happen much faster, even twice a year. Long term support releases will come every 3 years - but to get that three years support updates you will probably need to pay for it, too. See the blog link at end of article on free, secure, stable.
+
+JDK 11 will continue the modularity work and more aggressively remove modules from core Java, making it lighter and easier to update. Next on list are CORBA and Java EE modules. You can still get them, on top of core Java, when you need them. But they're not bundled with every micro-service, freeing up 22 megabytes and 9 modules from it. It goes leaner and meaner.
+
+Project Panama and Valhalla are still coming up, value types, generic specialization, native function calling from JVM, etc. Amber project started with local variable type inference, but enums will be having much more love in future versions, too. At this early stage, not much concrete to show yet.
+
+But it's probably safe to say - this is not your grandfather's Java anymore.
+
+## Disclaimer and closing thoughts
+
+You might have noticed that I'm an incurable jester. Half of what I speak is not so serious, trick is detecting which half. I do not endorse you to use always the latest available platform for running production systems, unless you know what you are doing. I certainly do not encourage you to run outdated ones, either. But you should be relatively fine with Java 8 for some years more. So please do not install JDK 11 on your production system yet.
+ 
+I do have great passion for technology and believe firmly in always having eye for future, and playing with things before using them. Java 11 is now in play-phase, and this article is about understanding you can already start learning it and playing with it, to be prepared. 
```

Comment:
"I do have **a** great…"

"Having **an** eye for **the** future"

"…using them *for real*" maybe?

"understanding **that** you…"